### PR TITLE
Bump rubocop version

### PR DESCRIPTION
**Pull request by rr3tt**
Bumping the rubocop version up to the most recent: https://github.com/rubocop/rubocop/releases

### CODE REVIEW

**Review by adam-hicks — APPROVED** [review]
any changes of note between the 2 rubocop versions?

**Comment by rr3tt** [comment]
I ran it against fantasy and it gave:
```
4840 files inspected, 272 offenses detected, 212 offenses autocorrectable
```
so will be a few fixes to go through in order to update fantasy to this

**Comment by ENewmeration** [comment]
> I ran it against fantasy and it gave:
> 
> ```
> 4840 files inspected, 272 offenses detected, 212 offenses autocorrectable
> ```
> 
> so will be a few fixes to go through in order to update fantasy to this

I think we need to block merging this then until those are updated otherwise it will block ALL PRs until then

**Review by ENewmeration — APPROVED** [review]
gonna block this until we make the changes which I think will require a bigger discussion depending on the stylistic additions. That or we ignore the new things for now

**Review by ENewmeration — CHANGES_REQUESTED** [review]
dang it meant to hit request changes 🤦

**Comment by rr3tt** [comment]
@ENewmeration I didn't think this change would get pulled into fantasy unless we run a bundle update on underdog ruby style, it should be pinned to the commit `1e217064d85bdbc34609d732f2573b7aa5ba159d` via the Gemfile.lock: https://github.com/Underdog-Inc/api/blob/main/Gemfile.lock#L29

**Comment by ENewmeration** [comment]
> > I think we need to block merging this then until those are updated otherwise it will block ALL PRs until then
> > Gemfile is pointing to main branch so it will autopull this when this is merged
> 
> @ENewmeration I didn't think this change would get pulled into fantasy unless we run a bundle update on underdog ruby style, it should be pinned to the commit `1e217064d85bdbc34609d732f2573b7aa5ba159d` via the Gemfile.lock: [Underdog-Inc/api@`main`/Gemfile.lock#L29](https://github.com/Underdog-Inc/api/blob/main/Gemfile.lock?rgh-link-date=2023-11-21T22%3A11%3A38Z#L29)

Looks like you are right it is using that exact revision in the build so I'll approve again https://github.com/Underdog-Inc/api/actions/runs/6948886816/job/18906643492?pr=5246#step:3:327

File: underdog_ruby_style.gemspec [hunk 1]
```diff
@@ -32,11 +32,9 @@ Gem::Specification.new do |spec|
 
   # For more information and examples about making a new gem, check out our
   # guide at: https://bundler.io/guides/creating_gem.html
-  spec.add_dependency "rubocop", "1.41.1"
+  spec.add_dependency "rubocop", "1.57.2"
   spec.add_dependency "rubocop-minitest"
   spec.add_dependency "rubocop-performance"
   spec.add_dependency "rubocop-rails"
-  spec.add_development_dependency "bundler", "~> 2.3"
-  spec.add_development_dependency "rake", "~> 13.0"
```
Comment (ENewmeration) [inline underdog_ruby_style.gemspec:40]: we still have these in the gemfile looks like, any reason they aren't required anymore in the gemspec?

File: Gemfile [hunk 2]
```diff
@@ -5,8 +5,7 @@ source "https://rubygems.org"
 # Specify your gem's dependencies in underdog_ruby_style.gemspec
 gemspec
 
-gem "rake", "~> 13.0"
-
+gem "bundler", "~> 2.3"
 gem "minitest", "~> 5.0"
-
+gem "rake", "~> 13.0"
 gem "rubocop", "~> 1.21"
```
Comment (ENewmeration) [inline Gemfile:11]: this rubocop can probably be removed right?

File: underdog_ruby_style.gemspec (hunk 1 above)
Comment (ENewmeration) [reply underdog_ruby_style.gemspec:40]: or these just needed for local build and not bundling of gem

**Review by ENewmeration — APPROVED** [review]

File: underdog_ruby_style.gemspec (hunk 1 above)
Comment (rr3tt) [reply underdog_ruby_style.gemspec:40]: it's a new rule, see the build of the first commit in this PR where it failed, here's some of the output:
```
underdog_ruby_style.gemspec:39:3: C: Gemspec/DevelopmentDependencies: Specify development dependencies in Gemfile.
  spec.add_development_dependency "bundler", "~> 2.3"
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
underdog_ruby_style.gemspec:40:3: C: Gemspec/DevelopmentDependencies: Specify development dependencies in Gemfile.
  spec.add_development_dependency "rake", "~> 13.0"
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

looks like there is an option in [Gemspec/DevelopmentDependencies](https://docs.rubocop.org/rubocop/cops_gemspec.html#gemspecdevelopmentdependencies) to enforce the reverse, where dev deps should be specified in the gemspec.  I don't really care either way so open to whatever ppl want

Comment (ENewmeration) [reply underdog_ruby_style.gemspec:40]: eh I like how you have it

File: Gemfile (hunk 2 above)
Comment (rr3tt) [reply Gemfile:11]: yep, will remove

Comment (rr3tt) [reply Gemfile]: https://github.com/Underdog-Inc/underdog_ruby_style/pull/10/commits/1878d8ca680c7a5eaafa220dac0afdd250f94e26

**Pull request merged**